src/Native/include/nncase/ntt/arch/aarch64/primitive_ops.h

```diff
@@ -0,0 +1,391 @@
+/* Copyright 2019-2021 Canaan Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include "../../primitive_ops.h"
+#include "arch_types.h"
+#include "arm_math.h"
+
+namespace nncase::ntt::ops {
+
+// unary op
+
+// abs
+// template <> struct abs<ntt::vector<float, 8>> {
+//     ntt::vector<float, 8> operator()(const ntt::vector<float, 8>& v) const
+//     noexcept
+//     {
+//         return (v);
+//     }
+// };
+
+// acos
+// template <> struct acos<ntt::vector<float, 8>> {
+//     ntt::vector<float, 8> operator()(const ntt::vector<float, 8>& v) const
+//     noexcept
+//     {
+//         return acos256_ps(v);
+//     }
+// };
+
+// acosh(v) = ln(v + sqrt(v^2 - 1)), v >= 1
+// template <> struct acosh<ntt::vector<float, 8>> {
+//     ntt::vector<float, 8> operator()(const ntt::vector<float, 8>& v) const
+//     noexcept
+//     {
+//         auto ones = _mm256_set1_ps(1.0f);
+//         return log256_ps(_mm256_add_ps(
+//             v, _mm256_sqrt_ps(_mm256_sub_ps(_mm256_mul_ps(v, v), ones))));
+//     }
+// };
+
+// asin
+// template <> struct asin<ntt::vector<float, 8>> {
+//     ntt::vector<float, 8> operator()(const ntt::vector<float, 8>& v) const
+//     noexcept
+//     {
+//         return asin256_ps(v);
+//     }
+// };
+
+// asinh(v) = ln(v + sqrt(v^2 + 1))
+// template <> struct asinh<ntt::vector<float, 8>> {
+//     ntt::vector<float, 8> operator()(const ntt::vector<float, 8>& v) const
+//     noexcept
+//     {
+//         auto ones = _mm256_set1_ps(1.0f);
+//         return log256_ps(_mm256_add_ps(
+//             v, _mm256_sqrt_ps(_mm256_add_ps(_mm256_mul_ps(v, v), ones))));
+//     }
+// };
+
+// ceil
+// template <> struct ceil<ntt::vector<float, 8>> {
+//     ntt::vector<float, 8> operator()(const ntt::vector<float, 8>& v) const
+//     noexcept
+//     {
+//         return _mm256_ceil_ps(v);
+//     }
+// };
+
+// cos
+template <> struct cos<ntt::vector<float, 8>> {
```

Comment:
Is there any performance difference between the packing of 4 and packing of 8?

Reply:
I haven't tested, I just support two optional Lanes.

Reply:
If there is no difference, I think there is no need to repeat these codes.

Reply:
removed redundant code